luaui/Widgets/gui_awards.lua

```diff
@@ -66,341 +163,152 @@ local function findPlayerName(teamID)
-		CowAward = createAward('cow', 1, Spring.I18N('ui.awards.didEverything'), white, awards.goldenCow, offset)
-		offset = offset + offsetAdd
-	end
-	-- make sure the other awards lines are at the bottom
-	local minOffset = 120 + (offsetAdd*3)
-	if offset < minOffset then
-		offset = minOffset
+local function putOtherAwardIntoModel(award, winnersTable)
+	local teamID = winnersTable[1].teamID
+	local score = calculateAwardScore(award, winnersTable[1].score)
+	if teamID < 0 then
+		return
 	end
-	OtherAwards = createAward('', 2, '', white, compoundAwards, offset)
-
-	drawAwards = true
-
-	-- don't show graph
-	Spring.SendCommands('endgraph 0')
+	local name = findPlayerName(teamID)
+	table.insert(modelForPresenter.otherAwards, {
+		visible = true,
+		award = award,
+		leader = {
+			playerName = name,
+			score = score,
+			playerColor = getCSSColorByPlayer(teamID),
+		},
+	})
 end
 
-function widget:MousePress(x, y, button)
-	if drawAwards then
-		if button ~= 1 then
-			return
-		end
-
-		-- Leave button
-		if (x > widgetX + widgetWidthScaled - quitRightX - math.floor(5*widgetScale)
-				and (x < widgetX + widgetWidthScaled - quitRightX + math.floor(20*widgetScale) * font:GetTextWidth(Spring.I18N('ui.awards.leave')) + math.floor(5*widgetScale))
-				and (y > widgetY + math.floor((50 - 5)*widgetScale))
-				and (y < widgetY + math.floor((50 + 17 + 5)*widgetScale))) then
-			if chobbyLoaded then
-				Spring.Reload("")
-			else
-				Spring.SendCommands("quitforce")
-			end
-		end
-
-		-- Show Graphs button
-		if (x > widgetX + widgetWidthScaled - graphsRightX - math.floor(5*widgetScale))
-				and (x < widgetX + widgetWidthScaled - graphsRightX + math.floor(20*widgetScale) * font:GetTextWidth(Spring.I18N('ui.awards.showGraphs')) + math.floor(5*widgetScale))
-				and (y > widgetY + math.floor((50 - 5)*widgetScale)
-					and (y < widgetY + math.floor((50 + 17 + 5)*widgetScale))) then
-			Spring.SendCommands('endgraph 2')
-
-			if WG['guishader'] then
-				WG['guishader'].RemoveRect('awards')
-			end
-			drawAwards = false
-		end
+local function createMainAward(award, winnersTable)
+	putMainAwardIntoModel(award, 1, winnersTable)
+	putMainAwardIntoModel(award, 2, winnersTable)
+	putMainAwardIntoModel(award, 3, winnersTable)
+end
 
-		-- Close button
-		if (x > widgetX + widgetWidthScaled - closeRightX - math.floor(5*widgetScale))
-				and (x < widgetX + widgetWidthScaled - closeRightX + math.floor(20*widgetScale) * font:GetTextWidth('X') + math.floor(5*widgetScale))
-				and (y > widgetY + widgetHeightScaled - math.floor((10 + 17 + 5)*widgetScale)
-				and (y < widgetY + widgetHeightScaled - math.floor((10 - 5)*widgetScale))) then
-			if WG['guishader'] then
-				WG['guishader'].RemoveRect('awards')
-			end
-			drawAwards = false
-		end
-	end
+local function createOtherAward(award, winnersTable)
+	putOtherAwardIntoModel(award, winnersTable)
 end
 
-function widget:DrawScreen()
-	if not drawAwards then
+local function ProcessAwards(awards)
```

Comment:
```suggestion
local function processAwards(awards)
```
Locals start with lowercase

Reply:
fixed